docs/structured-streaming-programming-guide.md

```diff
@@ -0,0 +1,888 @@
+---
+layout: global
+displayTitle: Structured Streaming Programming Guide [Alpha]
+title: Structured Streaming Programming Guide
+---
+
+* This will become a table of contents (this text will be scraped).
+{:toc}
+
+# Overview
+Structured Streaming is a scalable and fault-tolerant stream processing engine 
+built on the Spark SQL engine. You can express your streaming computation by 
+thinking you are running a batch computation on a static dataset, and the 
+Spark SQL engine takes care of running it incrementally and continuously 
+updating the final result as streaming data keeps arriving. You can use the 
+[Dataset/DataFrame API](sql-programming-guide.html) in Scala, Java or Python to express streaming 
+aggregations, event-time windows, stream-to-batch joins, etc. The computation 
+is executed on the same optimized Spark SQL engine. Finally, the system 
+ensures end-to-end exactly-once fault-tolerance guarantees through 
```

Comment:
End-to-end exactly-once sounds like over-promising.  Should probably define what the ends are, because destructive outputs can't be literally exactly-once in the face of network failures.